session/session_fail_test.go

```diff
@@ -22,17 +22,24 @@ import (
 )
 
 func (s *testSessionSuite) TestFailStatementCommit(c *C) {
-	defer gofail.Disable("github.com/pingcap/tidb/session/mockStmtCommitError")
 
 	tk := testkit.NewTestKitWithInit(c, s.store)
 	tk.MustExec("create table t (id int)")
 	tk.MustExec("begin")
 	tk.MustExec("insert into t values (1)")
+
 	gofail.Enable("github.com/pingcap/tidb/session/mockStmtCommitError", `return(true)`)
 	_, err := tk.Exec("insert into t values (2)")
 	c.Assert(err, NotNil)
-	tk.MustExec("commit")
-	tk.MustQuery(`select * from t`).Check(testkit.Rows("1"))
+
+	gofail.Disable("github.com/pingcap/tidb/session/mockStmtCommitError")
+
+	tk.MustExec("insert into t values (3)")
+	tk.MustExec("insert into t values (4)")
+	_, err = tk.Exec("commit")
+	c.Assert(err, NotNil)
+
+	tk.MustQuery(`select * from t`).Check(testkit.Rows())
```

Comment:
Please add a test that error occurred during the partial StmtCommit to Transaction, and make sure there is no data consistent problem.

Reply:
Now, line 32 mock partial of the data fail...